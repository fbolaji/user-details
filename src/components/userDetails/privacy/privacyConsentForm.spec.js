import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import PrivacyConsentForm from './privacyConsentForm';

import MutationObserver from 'mutation-observer';

let targetNode = document.getElementById('root');
let config = { attributes: true, childList: true, subtree: true, characterData:true };
const observer = new MutationObserver(function() {
	observer.disconnect();
});

describe("PrivacyConsentForm", () =>{
	it("should render form elements if data is > 0", () => {
		const userPrivacyForm = [
			{
				"checked": false,
				"label": "Recieve updates about Tray.io products by email",
				"id": "productUpdate",
				"name": "productUpdate",
				"type": "checkbox"
			},
			{
				"checked": false,
				"label": "Recieve communication about by email for other products created by Tray.io team",
				"id": "otherProductUpdate",
				"name": "otherProductUpdate",
				"type": "checkbox"
			}
		];

		const { container, getByText, getByTestId } = render(<PrivacyConsentForm
			userPrivacyForm={userPrivacyForm}
		/>);
		const testid = getByTestId('userPrivacyForm_testid');
		expect(testid).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});