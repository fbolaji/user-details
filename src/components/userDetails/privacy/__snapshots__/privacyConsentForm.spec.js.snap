// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyConsentForm should render form elements if data is > 0 1`] = `
<div>
  <form
    class=""
    data-testid="userPrivacyForm_testid"
    name="userPrivacyForm"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="productUpdate"
        name="productUpdate"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="productUpdate"
      >
        Recieve updates about Tray.io products by email
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="otherProductUpdate"
        name="otherProductUpdate"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="otherProductUpdate"
      >
        Recieve communication about by email for other products created by Tray.io team
      </label>
    </div>
    <div
      class="pt-4"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Back
      </button>
      <button
        class="float-right btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
